include/s3select_functions.h

```diff
@@ -846,6 +855,119 @@ struct _fn_version : public base_function
   }
 };
 
+struct _fn_isnull : public base_function
+{
+
+  value res;
+  
+  bool operator()(bs_stmt_vec_t* args, variable* result)
+  {
+    bs_stmt_vec_t::iterator iter = args->begin();
+    base_statement* expr = *iter;
+    value expr_val = expr->eval();
+    if ( expr_val.is_null()) {
+      result->set_value((int64_t)true);
+    } else {
+      result->set_value((int64_t)false);
+    }
+    return true;
+  }
+};
+
+struct _fn_in : public base_function
+{
+
+  value res;
+  
+  bool operator()(bs_stmt_vec_t* args, variable* result)
+  {
+    bs_stmt_vec_t::iterator iter = args->begin();
```

Comment:
please use meaningful and consistent variable names;  if you really need temporary references to begin() and end(), use something like iter_begin and iter_end.  "iter" and "it" are both conventions for an iterator where there can be no ambiguity about what it points to, such as the iterator in a loop over sequence;  but they are -different conventions-;  you don't use "it" and "iter" together;  I prefer iter--so dos Lakos :)